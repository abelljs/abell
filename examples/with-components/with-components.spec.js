/* eslint-disable max-len */
const fs = require('fs');
const path = require('path');

const {
  preTestSetup,
  getSelector,
  transformHash
} = require('../../tests/test-utils/helpers.js');

describe('examples/with-components', () => {
  beforeAll(async () => {
    await preTestSetup('with-components');
  });

  describe('index.html', () => {
    let $;
    beforeAll(() => {
      $ = getSelector(path.join(__dirname, 'dist', 'index.html'));
    });

    it('should render html from <Nav/>', () => {
      expect($('[data-test="nav-component"]').html()).toMatchSnapshot();
    });

    it('should render data from <Footer/>', () => {
      expect($('footer').text()).toBe('This is Footer');
    });

    it('should add main stylesheet link to head', () => {
      expect($('link[rel="stylesheet"]').attr('href')).toBe(
        'bundled-css/main.abell.css'.replace(/\//g, path.sep)
      );
    });

    it('should have inlined CSS in the index page', () => {
      expect(transformHash($('style').html())).toMatchSnapshot();
    });
  });

  describe('my-first-blog/index.html', () => {
    let $;
    beforeAll(() => {
      $ = getSelector(
        path.join(__dirname, 'dist', 'my-first-blog', 'index.html')
      );
    });

    it('should add main stylesheet link to head', () => {
      expect($('link[rel="stylesheet"]').attr('href')).toBe(
        '../bundled-css/main.abell.css'.replace(/\//g, path.sep)
      );
    });

    it('should have inlined CSS in the blog page', () => {
      expect(transformHash($('style').html())).toMatchSnapshot();
    });
  });

  describe('about.html', () => {
    let $;
    beforeAll(() => {
      $ = getSelector(path.join(__dirname, 'dist', 'about.html'));
    });

    it('should have inlined CSS in the about page', () => {
      expect(transformHash($('style').html())).toMatchSnapshot();
    });
  });

  describe('bundled-css/main.abell.css', () => {
    it('should have expected CSS properties in main bundle', () => {
      const footerCSS = `
        footer[data-abell-jLzQID] {
          color: #f30;
        }
      `.replace(/\s|\n|\r/g, '');

      const aboutCSS = `
        .about[data-abell-fBMCHA] {
          color: #999;
        }
      `.replace(/\s|\n|\r/g, '');

      const mainAbellCSS = fs
        .readFileSync(
          path.join(__dirname, 'dist', 'bundled-css', 'main.abell.css'),
          'utf-8'
        )
        .replace(/\s|\n|\r/g, '');

      expect(
        mainAbellCSS.includes(footerCSS) && mainAbellCSS.includes(aboutCSS)
      ).toBe(true);
    });
  });

  describe('bundled-js/main.abell.js', () => {
    it('should have expected JS code in main js bundle', () => {
      expect(
        fs
          .readFileSync(
            path.join(__dirname, 'dist', 'bundled-js', 'main.abell.js'),
            'utf-8'
          )
          .replace(/\s|\n|\r/g, '')
      ).toBe(
        `document.querySelector('div.brand').innerHTML = 'Set from JS';`.replace(
          /\s|\n|\r/g,
          ''
        )
      );
    });
  });
});
