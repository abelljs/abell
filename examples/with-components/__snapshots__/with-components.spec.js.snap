// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/with-components about.html should have inlined CSS in the about page 1`] = `"nav[data-abell-1]{background-color:#000;color:#fff;}"`;

exports[`examples/with-components index.html should have inlined CSS in the index page 1`] = `"nav[data-abell-1]{background-color:#000;color:#fff;}"`;

exports[`examples/with-components index.html should render html from <Nav/> 1`] = `
"
    <div data-test=\\"brand-component\\" class=\\"brand\\" data-abell-fmljlv data-abell-iycgar>This is brand component</div>
    This is nav component <span data-test=\\"nav-component-prop\\" data-abell-iycgar>3</span>
  "
`;

exports[`examples/with-components my-first-blog/index.html should have inlined CSS in the blog page 1`] = `"nav[data-abell-1]{background-color:#000;color:#fff;}"`;
